### PR TITLE
Feature/battle

### PR DESCRIPTION
**Pull request by TheGreatAxios**
#1 Feature/battle

## Overview
Contains the core battle smart contract, minor tweaks and functions to the GameItems, core Unit Testing for Battle. 

- Also started the integration tests for confirmation of successful battle contract

### CODE REVIEW

File: contracts/contracts/Battle.sol [hunk 1]
```diff
@@ -5,27 +5,222 @@ import "@openzeppelin/contracts/token/ERC1155/ERC1155.sol";
...
+    _removeCommitment(opponent);
+    /// 6 - Emit Battle Event
+    emit BattleCompleted(msg.sender, opponent, result, _p1TokenId, _p2TokenId); /// TO DO FIX
   }
 
-  // this 
-  mapping(address => mapping(uint => PlayedItem)) public rounds;
+  /// @dev Enables Player to Commit an Item
+  /// @param commitment Takes a hashed commit and stores on chain
+  function commitItem(bytes32 commitment, uint256 tokenId) external {
```
Comment (tobowers) [inline contracts/contracts/Battle.sol:121]: calling this function means that tokenId is public

File: contracts/contracts/Battle.sol [hunk 2]
```diff
@@ -5,27 +5,222 @@ import "@openzeppelin/contracts/token/ERC1155/ERC1155.sol";
...
+
+    IGameItems _gameItemsContract = _buildGameItems();
+
+    /// Loads Number of Items to Loop Through
+    uint256 _numberItems = _gameItemsContract.getNumberItems();
+    uint256 _p1TokenId = 1000; /// Initial Setting of MyTokenId
+    uint256 _p2TokenId = 1000; /// Initial Setting of Opponent Token Id
+
+    /// Loops Through # of Items
+    for (uint i = 0; i < _numberItems; i++) {
```
Comment (tobowers) [inline contracts/contracts/Battle.sol:86]: we shouldn't need to loop through anything.

**Review by tobowers — APPROVED** [review]
i'm going to approve and merge so we can work *from* here, but I think the commitment scheme is broken (because tokenID becomes public).

I think what we want is a commitment is a pure byte32... it is keccak256(salt, tokenID). The reveal is salt and tokenID (which then gets matched to commitment). We *then* check to make sure the player has the token. If they don't then their account is effectively frozen (and the only way they could do that is with trying to hack it).

File: contracts/contracts/Battle.sol (hunk 1 above)
Comment (TheGreatAxios) [reply contracts/contracts/Battle.sol:121]: No worries, will remove. Thanks for clarifying; I was unsure on that front.

**Pull request merged**